setup/setup.iss

```diff
@@ -362,16 +369,33 @@ begin
         if not CreateSymbolicLinkA(GetGameDir('RedFaction.exe'), ExpandConstant('{app}\DashFactionLauncher.exe'), 0) then
         begin
             Log('CreateSymbolicLink failed');
-            MsgBox('Failed to replace Red Faction launcher by a symbolic link to Dash Faction launcher.', mbError, MB_OK);
+            MsgBox('Failed to replace the Red Faction launcher with a symbolic link to the Dash Faction launcher.', mbError, MB_OK);
+        end;
+    end;
+end;
+
+procedure RestoreRedFactionLauncher;
+var 
+   GamePath: String;
+begin
+    // Use the registry entry to find the game directory
+    RegQueryStringValue(HKCU, 'Software\Volition\Red Faction\Dash Faction', 'Executable Path', GamePath);
+    GamePath := ExtractFileDir(GamePath) + '\';
+    if FileExists(GamePath + 'RedFaction.exe.bak') then
+    begin
+        if FileExists(GamePath + 'RedFaction.exe') then
+        begin
+            DeleteFile(GamePath + 'RedFaction.exe');
         end;
+        RenameFile(GamePath + 'RedFaction.exe.bak', GamePath + 'RedFaction.exe');
     end;
 end;
 
 // Check functions
 
 function PatchGameTaskCheck(): Boolean;
 begin
-    Result := Length(Patches) > 0;
+    Result := GetArrayLength(Patches) > 0;

```

Comment:
hm, according to Inno documentation `Length` and `SetLength` work on strings. Did it work before this change? :O

Reply:
It did not work for the version of Inno I was using. Looks like `Patches` is actually an array. Does it work for you?

Reply:
Are you using Inno 6.x? I am usually building with Inno 5.x and it it must have worked because I tested patching multiple times. But as I mentioned even Inno 5.x says that those functions are for strings. I think I used them because AFAIK that's how you work with arrays in Pascal (https://www.freepascal.org/docs-html/rtl/system/length.html). Maybe Inno differs here...

Reply:
I was using Inno 5.6.1 and going off of their own documentation:
https://jrsoftware.org/ishelp/index.php?topic=isxfunc_getarraylength
https://jrsoftware.org/ishelp/index.php?topic=isxfunc_setarraylength